code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -101,7 +101,7 @@
 	taste_description = "slime"
 
 /datum/reagent/vaccine/reaction_mob(mob/living/L, method=TOUCH, reac_volume)
-	if(islist(data) && (method == INGEST || method == INJECT))
+	if(!islist(data) || !(method & INJECT) || reac_volume < 5)
```

Comment:
You inverted the check. Now vaccines only cure diseases if they don't have any diseases to cure, are specifically not injected, and have less than 5 units.